### PR TITLE
refinements to time comparison toggle behavior

### PR DESCRIPTION
**Pull request by bcolloran**
addresses feedback under this section: https://www.notion.so/rilldata/Design-Rev-on-Display-percent-of-total-metrics-in-dashboards-2bbf36511ff9452d949286c5a996205e?pvs=4#f086979733b8484ebaccfb41e37b7878 (except for tooltip string change, which is in #2388)

### CODE REVIEW

**Comment by bcolloran** [comment]
@AdityaHegde requesting your review on this. 

Also may need some help getting it over the line with the state serialization in the protobufs. I tried just following what you did in https://github.com/rilldata/rill/pull/2772 since that PR was so straightforward, but when I tried to regenerate the bindings I got a ton of changes in several files, not just the small set of changed you got in `web-common/src/proto/gen/rill/ui/v1/dashboard_pb.ts` in #2772. I'll keep looking into it (might be down to a different version of `buf` on my system), but if I can't get it sorted out I might ask you to finish up that last bit.

**Comment by bcolloran** [comment]
It looks like it's not just the version of `buf`, but also `orval`.  I think it's probably not smart for me to spend more time on this since I'm not familiar with this and don't want to break anything -- I thought I'd try to do it if I could just replicate what you got in #2772, but since it's turned out to be more than trivial to wire up the URL state, I'd be most appreciative if you could take that part @AdityaHegde. I suspect it will be quick for you :-)

File: web-common/src/features/dashboards/dashboard-stores.ts [hunk 1]
```diff
@@ -61,8 +61,10 @@ export interface MetricsExplorerEntity {
   // flag to show/hide comparison based on user preference
   showComparison?: boolean;
 
-  // flag to show/hide the percent of total column
+  // flag to show/hide the percent of total column *in the leaderboard*
   showPercentOfTotal?: boolean;
+  // flag to show/hide the delta change column *in the leaderboard*
+  showDeltaChange?: boolean;
```
[the remark this answers is not in the document]
Comment (bcolloran) [reply web-common/src/features/dashboards/dashboard-stores.ts]: updated to use enum

**Comment by bcolloran** [comment]
Hey @AdityaHegde, I updated this PR to follow your suggestion of using an enum, so it should be ready for you to take if from here for the URL state. One note: changing to an enum involved touching a few extra bits of code, so please pay close attention to make sure the behaviors are still sensible and desired when doing your QA pass for the URL part. Thanks!

**Comment by bcolloran** [comment]
@AdityaHegde this PR is getting stale, can you fix the conflicts and merge?

for `dashboard-stores`, the updated section should be:
```javascript
  // user selected timezone
  selectedTimezone?: string;

  // flag to show/hide time comparison based on user preference.
  // This controls whether a time comparison is shown in e.g.
  // the line charts and bignums.
  // It does NOT affect the leaderboard context column.
  showComparison?: boolean;
```

For `LeaderboardDisplay`, we can take the new code from `main`

(I tried to do this myself, but I cannot resolve just those merge conflicts without committing conflicted versions of the other files)

This is a soft blocker for #2826.

**Review by AdityaHegde — APPROVED** [review]

**Pull request merged**